localstack/runtime/init.py

```diff
@@ -152,8 +152,12 @@ def run_stage(self, stage: Stage) -> List[Script]:
                 else:
                     script.state = State.SUCCESSFUL
                 finally:
-                    os.environ.clear()
-                    os.environ.update(env_original)
+                    # Restore original state of Boto credentials.
+                    for env_var in ("AWS_ACCESS_KEY_ID", "AWS_SECRET_ACCESS_KEY", "AWS_REGION"):
+                        if env_var in env_original:
+                            os.environ[env_var] = env_original[env_var]
```

Comment:
I'm wondering whether a modification to the value made by the init hook should be reset or retained?

Reply:
I think reset is honestly fine, as you might just use them for the operations inside the init hook script.